metrics/repl.lag/lag.go

```diff
@@ -115,70 +130,116 @@ func (c *Lag) Help() blip.CollectorHelp {
 	}
 }
 
+// Prepare prepares lag collectors for all levels in the plan that contain the "repl.lag" domain
 func (c *Lag) Prepare(ctx context.Context, plan blip.Plan) (func(), error) {
 LEVEL:
-	for _, level := range plan.Levels {
+	for levelName, level := range plan.Levels {
 		dom, ok := level.Collect[DOMAIN]
 		if !ok {
 			continue LEVEL // not collected in this level
 		}
+		c.dropNotAReplica[levelName] = !blip.Bool(dom.Options[OPT_REPORT_NOT_A_REPLICA])
+		c.replCheck = sqlutil.CleanObjectName(dom.Options[OPT_REPL_CHECK]) // @todo sanitize better
 
-		table := dom.Options[OPT_TABLE]
-		if table == "" {
-			table = blip.DEFAULT_HEARTBEAT_TABLE
+		if source, ok := dom.Options[OPT_LAG_SOURCE]; ok {
+			if len(source) > 0 && source != "auto" {
+				switch source {
```

Comment:
Makes sense, I will update. We use [a similar form](https://github.com/cashapp/blip/blob/29f6a8032581516c7359364c4df27f858e47910a/metrics/var.global/global.go#L129-L149) (I based this off that) to this in `var.global` collector, we should refactor that as well when we touch it next time.

Reply:
Refactored it, it has different behavior though.

Previously, tests were passing with just:

```
    repl.lag:
      options: {}
```
I didn't have to explicitly, set `writer: auto`. Even though, `if source, ok := dom.Options[OPT_LAG_SOURCE]; ok {` was returning `false` (I am not sure if it's a test only problem or it's how options are parsed, I am not too familiar with options parsing), as the default case was outside, it was still setting the correct source.

Now, without setting `writer: auto` it fails in collection because the that condition failing and default of `auto` is not being handled. But we were still returning `nil, nil` signalling everything is alright.

I check the implementation of [var.global](https://github.com/cashapp/blip/blob/29f6a8032581516c7359364c4df27f858e47910a/metrics/var.global/global.go#L147-L165) which seems to be handling default case separately as well, so maybe that's by design?